File: daily/2025/2523_Closest_Prime_Numbers_in_Range.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    bool isPrime(int num){
        int i = 2;
        while(i * i <= num){
            if (num % i == 0) return false;
            i++;
        }
        return true;
    }
    vector<int> closestPrimes(int left, int right) {
        vector<int> ans = {-1, -1};
        int diff = INT_MAX;
        int prev = 0;
        for (int curr = max(2, left); curr <= right; curr++){
            if (isPrime(curr)){
                if (prev && curr - prev < diff){
                    diff = curr - prev;
                    ans = {prev, curr};
                }
                prev = curr;
            }
        }
        return ans;
    }
};

int main(){
    int left = 10;
    int right = 19;
    // int left = 4;
    // int right = 6;
    Solution S;
    vector<int> ans = S.closestPrimes(left, right);
    cout << ans[0] << " " << ans[1] << endl;
    return 0;
}

File: daily_python/2523_Closest_Prime_Numbers_in_Range.py
import os
import sys
from typing import List

class Solution:
    def closestPrimes(self, left: int, right: int) -> List[int]:
        isPrime = [1] * (right + 1)
        isPrime[0] = 0
        isPrime[1] = 0
        idx = 2
        while idx * idx <= right:
            if isPrime[idx] == 1:
                for multiple in range(idx*idx, right+1, idx):
                    isPrime[multiple] = 0
            idx += 1
        prime = [num for num in range(left, right+1) if isPrime[num]]
        
        if len(prime) < 2:
            return (-1, -1)
        diff = float('inf')
        for i in range(1, len(prime)):
            if prime[i] - prime[i-1] < diff:
                diff = prime[i] - prime[i-1]
                ans = (prime[i-1], prime[i])
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            left = int(lines[0].split('=')[1].strip()[:-1])
            right = int(lines[1].split('=')[1].strip()[:-1])
            dataset.append((left, right))
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for left, right in dataset:
        results.append(solution.closestPrimes(left, right))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')